streaming/src/main/scala/org/apache/spark/streaming/Checkpoint.scala

```diff
@@ -139,9 +139,11 @@ class CheckpointWriter(
           // Write checkpoint to temp file
           fs.delete(tempFile, true)   // just in case it exists
           val fos = fs.create(tempFile)
-          fos.write(bytes)
-          fos.close()
-
+          try {
```

Comment:
Look at the new `Utils.tryWithSafeFinally` method which is used in a lot of the same contexts in the code now. It prevents an exception on closing a writer from masking the original exception.